tools/android/packaging/xbmc/src/org/xbmc/kodi/Main.java.in

```diff
@@ -182,6 +183,47 @@ public class Main extends NativeActivity implements Choreographer.FrameCallback
               | View.SYSTEM_UI_FLAG_HIDE_NAVIGATION
               | View.SYSTEM_UI_FLAG_FULLSCREEN
               | API_SYSTEM_UI_FLAG_IMMERSIVE_STICKY);
+
+
+      if (getPackageManager().hasSystemFeature("android.software.leanback"))
+      {
+        // Leanback
+
+        new Thread()
+        {
+          public void run()
+          {
+            try
+            {
+              sleep(10000);
+              mJsonRPC.updateLeanback(Main.this);
+            } catch (Exception e)
+            {
+              e.printStackTrace();
+            }
+          }
+        }.start();
```

Comment:
Well  #1 would be true if there was a way to have HTTP server only on localhost :( Users are not really aware of the risk of the webserver with bad configuration like possible exposing of their passwords.

Securing WebServer / JSON was long discussed but never approached :(